fe/lib/components/JobCategorySelect/JobCategorySelect.tsx

```diff
@@ -27,6 +33,7 @@ interface Props extends Omit<FieldProps, 'value' | 'onChange' | 'children'> {
     type: JobCategoryType,
   ) => void;
   schemeId: string;
+  initialValue?: { parent?: string | undefined; child?: string | undefined };
```

Comment:
I'm a bit suspect of this interface:

- What happens if the `parent` & `child` don't match?
- This pushes complexity on the caller to look up the correct `parent` & `child`

Would it be better to have an `initialJobCategoryId` and then figure out if it's a parent or child category internally?

Reply:
yeah I think the caller (i.e `PositionProfileCreator`) only has a child JobCategoryId (unless we include parent in the PositionProfile query), in addtion, we also query list of all categories in `JobCategorySelect`, we would only need one input for a selected category Id and can look up from there

Reply:
So we would pass down the `initialJobCategoryId` which is essentially child category Id and then do a look up to find the right parent Id? 

I did go down that path in this commit: https://github.com/seek-oss/wingman/pull/512/commits/6395f03ae9e48c81880bde28cdb7af34327d0e42 (Please ignore the horrible nested for loop. I always intended to replace that)

Reply:
You should be able to determine if `initialJobCategoryId` is a parent or child once you look it up in the list of job categories. From there you can handle it appropriately.

Reply:
@etaoins and @phongtcao let me know if this solves your concern. 

> From there you can handle it appropriately - either just set a parent, or look up a child’s parent.

@etaoins from what I can see we will always pass down child id as job category id. Did you want us to solve for a scenario where only parent id is passed down? In that case we just set the parent id and keep the child selection empty?

Reply:
It'd be good to support parent IDs as well. This a common open source component that Ryanair happens to use; we should make it generalisable to other use cases.